unicodetools/src/main/resources/org/unicode/text/UCD/UnicodeInvariantTest.txt

```diff
@@ -446,22 +446,44 @@ $Unicode_6_1_Decompositions ⊆ [\p{Name=/^CHAKMA VOWEL SIGN /}]
 # Stability: All characters other than those with General_Category property values Spacing_Mark (Mc) and Nonspacing_Mark (Mn) have the Canonical_Combining_Class property value 0.
 \p{CCC=0} ⊇ [^ \p{GC=Mc} \p{GC=Mn}]
 
-# Stability: Canonical and compatibility mappings (Decomposition_Mapping property values) are always in canonical order, and the resulting recursive decomposition will also be in canonical order. 
-# Note: We really mean the fixed point of Decomposition_Mapping on the left-hand side here.
-In \P{Decomposition_Mapping=}, Decomposition_Mapping * Decomposition_Mapping * Decomposition_Mapping = toNFKD
-
-# Stability: Canonical mappings (Decomposition_Mapping property values) are always limited either to a single value or to a pair. The second character in the pair cannot itself have a canonical mapping. 
+# Stability: Canonical and compatibility mappings (Decomposition_Mapping
+# property values) are always in canonical order, and the resulting recursive
+# decomposition will also be in canonical order. 
+# Note: We really mean the fixed point of Decomposition_Mapping on the
+# left-hand side here.
+In $codepoints, Decomposition_Mapping * Decomposition_Mapping * Decomposition_Mapping = toNFKD
+
+# Stability: Canonical mappings (Decomposition_Mapping property values) are
+# always limited either to a single value or to a pair. The second character in
+# the pair cannot itself have a canonical mapping.
 [ \p{Decomposition_Type=Canonical} & \p{Decomposition_Mapping=/.../} ] = []
-# TODO: The second character in the pair cannot itself have a canonical mapping.
-
-# Stability: Canonical mappings (Decomposition_Mapping property values) are always limited so that no string when normalized to NFC expands to more than 3× in length (measured in code units). 
+[ \p{Decomposition_Type=Canonical} & \p{Decomposition_Mapping=} ] = []
+Let $expandingCanonicalDecompositions = [ \p{Decomposition_Type=Canonical} & \p{Decomposition_Mapping=/../} ]
+In $expandingCanonicalDecompositions, Decomposition_Type * (drop 1) * Decomposition_Mapping ≠ (constant Canonical)
+# In fact the second character does not have a decomposition mapping at all,
```

Comment:
I think we need to make clear in the comments what are real failures, and what happens to be the case — so we want to be notified if it breaks — but is not a true invariant failure.

Reply:
Reworded to clarify.